lib/kitchen/driver/ec2.rb

```diff
@@ -167,6 +167,15 @@ def default_public_ip_association
         !!config[:subnet_id]
       end
 
+      def default_aws_session_token
+        if config[:aws_secret_access_key] == iam_creds[:aws_secret_access_key] \
+          && config[:aws_access_key_id] == iam_creds[:aws_access_key_id]
+          iam_creds[:aws_session_token]
+        else
```

Comment:
Actually I'm just checking that config[:aws_secret_access_key] et all had fallen back to the iam_creds statement so that we know it would be logical to return the iam_creds[:aws_session_token] since our other config values are from iam.